felix/bpf/libbpf/libbpf.go

```diff
@@ -492,6 +485,20 @@ func XDPSetGlobals(
 	return err
 }
 
+func SetGlobalData(m *Map, data interface{}) error {
+	switch v := data.(type) {
+	case *TcGlobalData:
+		return tcSetGlobals(m, v)
```

Comment:
Why not to have this function to be a method of each of the globals structs and then define an interface with that  method (if need be) and call that method rather than using _any_ and this switch?